src/IO/H5/Python/ExtractDatFromH5.py

```diff
@@ -4,28 +4,36 @@
 # See LICENSE.txt for details.
 
 from spectre.IO import H5 as spectre_h5
+import click
 import spectre.DataStructures
 import numpy as np
 import os
 import shutil
 
 
-def extract_dat_files(filename, **kwargs):
+def extract_dat_files(filename, out_dir, list=False, force=False):
+    """Extract dat files from an H5 file
+
+    Extract all Dat files inside a SpECTRE HDF5 file. The resulting files will
+    be put into the 'OUT_DIR'. The directory structure will be identical to the
+    group structure inside the HDF5 file.
+    """
     h5file = spectre_h5.H5File(filename, "r")
 
     all_dat_files = h5file.all_dat_files()
 
-    if kwargs["list"]:
+    if list:
         print_str = "\n ".join(all_dat_files)
         print("Dat files within '{}':\n {}".format(filename, print_str))
         return
 
-    out_dir = kwargs["output_directory"]
-
+    if out_dir is None:
+        raise ValueError(
+            "An output directory is required unless listing file content.")
```

Comment:
Can we keep it so that if you don't specify an output directory, it defaults to `extracted_NameOfH5File`?

Reply:
We could! I should have made that a separate commit and mentioned it in the description, sorry. I think output file/dir names should never(?) be defaults in a CLI, because then you run the script and don't know what happened / where to look for any output. It's super quick and easy to put the name of the output dir that you want as the last argument, and then you know exactly where output files are going. Added a note to the PR description.

Reply:
Yeah that's ok. I still think it's alright for there to be a default, but not worth holding up the CLI PRs